### ISSUE TITLE
Invalid JS output on Increment/Decrement static field

### ISSUE THREAD

**Issue by AndreyZM** [bridgedotnet/Bridge#1832]
Result:

```js
Uncaught ReferenceError: $t is not defined
```

### Expected

```js
Bridge.assembly("Demo", function ($asm, globals) {
    "use strict";

    Bridge.define("Demo.IncTest", {
        statics: {
            id_counter: 0,
            id: 0,
            config: {
                init: function () {
                    this.id = ((var $t = (Demo.IncTest.id_counter + 1) | 0, Demo.IncTest.id_counter = $t, $t));
                }
            }
        }
    });

    Bridge.define("Demo.Programm", {
        $main: function () {
            Bridge.Console.log(Demo.IncTest.id);
        }
    });
});
```

### Actual

```js
Bridge.assembly("Demo", function ($asm, globals) {
    "use strict";

    Bridge.define("Demo.IncTest", {
        statics: {
            id_counter: 0,
            id: 0,
            config: {
                init: function () {
                    this.id = (($t = (Demo.IncTest.id_counter + 1) | 0, Demo.IncTest.id_counter = $t, $t));
                }
            }
        }
    });

    Bridge.define("Demo.Programm", {
        $main: function () {
            Bridge.Console.log(Demo.IncTest.id);
        }
    });
});
```

### Steps To Reproduce

http://deck.net/de866c18a90cba2cb6ce0e5e85a5318e

```csharp
public class IncTest
{
    static public int id_counter = 0;
    static public int id = ++id_counter;
}

public class Programm
{
    static public void Main()
    {
        Console.WriteLine(IncTest.id);
    }
}
```

**Comment by LeonidVeriga** [comment]
### 1.14
Results in compiler exception:
```
Error: Object reference not set to an instance of an object.    at Bridge.Translator.AbstractEmitterBlock.GetTempVarName()
   at Bridge.Translator.AssignmentBlock.VisitAssignmentExpression()
   at Bridge.Translator.AssignmentBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitAssignmentExpression(AssignmentExpression assignmentExpression)
   at ICSharpCode.NRefactory.CSharp.AssignmentExpression.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.ParenthesizedBlock.EmitConversionExpression()
   at Bridge.Translator.ConversionBlock.DoEmit()
   at Bridge.Translator.AbstractEmitterBlock.Emit()
   at Bridge.Translator.Emitter.VisitParenthesizedExpression(ParenthesizedExpression parenthesizedExpression)
   at ICSharpCode.NRefactory.CSharp.ParenthesizedExpression.AcceptVisitor(IAstVisitor visitor)
   at Bridge.Translator.FieldBlock.WriteObject(String objectName, List`1 members, String format)
   at Bridge.Translator.FieldBlock.EmitFields(TypeConfigInfo info)
   at Bridge.Translator.FieldBlock.DoEmit()
```

### 15.0.0
Generates the same incorrect JavaScript for the static field initializer
```js
this.id = (($t = (IncTest.id_counter + 1) | 0, IncTest.id_counter = $t, $t));
``` 
```
Uncaught ReferenceError: $t is not defined
```

**Comment by LeonidVeriga** [comment]
Thanks for reporting @AndreyZM.
We will fix this issue in `15.1`

**Comment by LeonidVeriga** [comment]
@AndreyZM, a workaround is to use static constructor ([Deck](http://deck.net/b6ceacbe2cb54a337eb2a6bd69d91501)):
```cs
public class IncTest
{
    static public int id_counter = 0;
    static public int id;
    
    static IncTest()
    {
        id = ++id_counter;
    }
}

public class Programm
{
    static public void Main()
    {
        Console.WriteLine(IncTest.id);
        Console.WriteLine(IncTest.id_counter);
    }
}
```

**Comment by LeonidVeriga** [comment]
@AndreyZM, the fix will be included into **15.0.2**.

**Comment by geoffreymcgill** [comment]
This issue has been fixed and included in the Bridge 15.0.2 release that was pushed to NuGet. 

It will also be included in the upcoming Bridge 15.1.0 release. 

Your original Deck now compiles and runs correctly. 

Hope this helps.
